misaka/api.py

```diff
@@ -225,3 +259,112 @@ def __init__(self, nesting_level=6):
 
     def _new_renderer(self, flags, nesting_level):
         return lib.hoedown_html_toc_renderer_new(nesting_level)
+
+
+class SaferHtmlRenderer(HtmlRenderer):
+    """
+    A subclass of :class:`HtmlRenderer` which adds protections against
+    Cross-Site Scripting (XSS):
+
+    1. The ``'skip-html'`` flag is turned on by default, preventing injection of
+       HTML elements. If you want to escape HTML code instead of removing it
+       entirely, change ``sanitization_mode`` to ``'escape'``.
+    2. The URLs of links and images are filtered to prevent JavaScript injection.
+       See the :meth:`check_link` method below.
+    3. Optionally, the URLs can also be rewritten to counter other attacks such
+       as phishing.
+
+    Enabling URL rewriting requires extra arguments:
+
+    :arg link_rewrite: the URL of a redirect page, necessary to rewrite the
+        ``href`` attributes of links
+    :arg img_src_rewrite: the URL of an image proxy, necessary to rewrite the
+        ``src`` attributes of images
+
+    Both srings should include a ``{link}`` placeholder for the URL-encoded
+    target. Examples::
+
+        link_rewrite='https://example.com/redirect?url={link}',
+        img_src_rewrite='https://img-proxy-domain/{link}'
+    """
+    _allowed_url_re = re.compile(r'^https?:', re.I)
+
+    def __init__(self, flags=(), sanitization_mode='skip-html', nesting_level=0,
+                 link_rewrite=None, img_src_rewrite=None):
+        if not isinstance(flags, tuple):
```

Comment:
I didn't add support for passing integers because I saw that [it's deprecated](https://github.com/FSX/misaka/blob/c457f72df03f327be9b4077faf1d89fdb5f02bbc/misaka/utils.py#L47).